domain/application/service/unit.go

```diff
@@ -413,6 +418,36 @@ func (s *Service) GetUnitPrincipal(ctx context.Context, unitName coreunit.Name)
 	return s.st.GetUnitPrincipal(ctx, unitName)
 }
 
+// GetUnitMachineName gets the name of the unit's machine. If the unit's machine
+// cannot be found [applicationerrors.UnitMachineNotAssigned] is returned.
+func (s *Service) GetUnitMachineName(ctx context.Context, unitName coreunit.Name) (machine.Name, error) {
+	if err := unitName.Validate(); err != nil {
+		return "", errors.Capture(err)
+	}
+
+	unitMachine, err := s.st.GetUnitMachine(ctx, unitName)
```

Comment:
Please check the model is IAAS here

Reply:
The check should be done in the API facade. I don't want if checks here.

Reply:
This check is done in the uniter before the facade is called. Is there much point doing it in the facade? We will return a `NotAssigned` error in the case that it is called on IAAS which is reasonable behaviour.

In my opinion there is nothing to be gained from adding in an extra check for this. Feel free to convince me otherwise.

Reply:
The uniter may change, start calling this endpoint in new or different places, the uniter check may be lost in future.

However, returning `NotAssigned` is a resonable substitute for 'wrong model type'